core/txpool/txpool.go

```diff
@@ -241,9 +253,16 @@ func (p *TxPool) loop(head *types.Header, chain BlockChain) {
 		}
 		// Wait for the next chain head event or a previous reset finish
 		select {
+		case event := <-newOptimisticHeadCh:
+			if p.auctioneerEnabled {
+				// Chain moved forward, store the head for later consumption
+				newHead = event.Block.Header()
+			}
 		case event := <-newHeadCh:
-			// Chain moved forward, store the head for later consumption
-			newHead = event.Block.Header()
+			if !p.auctioneerEnabled {
+				// Chain moved forward, store the head for later consumption
+				newHead = event.Block.Header()
+			}
```

Comment:
why do we ignore a new head if auctioneer is enabled?

Reply:
yes, for auctioneer we only want to use optimistic blocks to validate the mempool state and not use the head block. that is the required functionality of auctioneer, to optimistically validate txs tip fee payment.  note that the optimistic block is always one block forward.